src/ShopifyClient.php

```diff
@@ -449,7 +449,7 @@ private function validateConnectionOptions(array $connectionOptions)
      */
     private function createDefaultClient(): GuzzleClient
     {
-        $baseUri = 'https://' . str_replace('.myshopify.com', '', $this->connectionOptions['shop']) . '.myshopify.com';
+        $baseUri = 'https://' . $this->connectionOptions['shop'];
```

Comment:
This is a breaking change and cannot be merged as it as it would make fail any people using it without appending the .myshopify.com in the shop option. Please make sure to handle that in a non breaking way :).